subsys/fs/nvs/nvs.c

```diff
@@ -175,11 +175,44 @@ static int nvs_flash_ate_wrt(struct nvs_fs *fs, const struct nvs_ate *entry)
 }
 
 /* data write */
-static int nvs_flash_data_wrt(struct nvs_fs *fs, const void *data, size_t len)
+static int nvs_flash_data_wrt(struct nvs_fs *fs, const void *data, size_t len, bool compute_crc)
 {
 	int rc;
 
-	rc = nvs_flash_al_wrt(fs, fs->data_wra, data, len);
+	/* Only add the CRC if required (ignore deletion requests, i.e. when len is 0) */
+	if (IS_ENABLED(CONFIG_NVS_DATA_CRC) && compute_crc && (len > 0)) {
+		size_t aligned_len, data_len = len;
+		uint8_t *data8 = (uint8_t *)data, buf[NVS_BLOCK_SIZE + NVS_DATA_CRC_SIZE], *pbuf;
+		uint32_t data_crc;
+
+		/* Write as much aligned data as possible, so the CRC can be concatenated at
+		 * the end of the unaligned data later
+		 */
+		aligned_len = len - (len % fs->flash_parameters->write_block_size);
```

Comment:
This can be simplified to: `aligned_len = len & ~(fs->flash_parameters->write_block_size - 1U);`

Reply:
Done !